example/ipsec_api/odp_ipsec.c

```diff
@@ -109,6 +109,10 @@ static void usage(char *progname);
 #define SHM_PKT_POOL_BUF_SIZE  4096
 #define SHM_PKT_POOL_SIZE      (SHM_PKT_POOL_BUF_COUNT * SHM_PKT_POOL_BUF_SIZE)
 
+/* Test stream input packet size */
```

Comment:
This does not appear to be the actual packet size but the (initial) size of the packet buffer that then gets filled with the packet of whatever length it happens to have.

The IPv4 packet creation function still looks fishy: at one point ICMP data of stream->length bytes gets written into the packet. I suppose the length could exceed the fixed packet buffer size as there is no check and as the length appears to come from the user. I also did not see any check that the packet data is actually contiguous in memory.

Reply:
Yes, it's the initial packet length. Now when you mentioned it, I finally found the undocumented '-s' command line argument, which can be used to select the test packet payload length. Will post V2.